packages/wdio-runner/tests/index.test.js

```diff
@@ -134,6 +134,17 @@ describe('wdio-runner', () => {
             expect(hook).toBeCalledTimes(0)
         })
 
+        it('should do nothing when triggered by run method with session that does not exist anymore', async () => {
+            const runner = new WDIORunner()
+            runner._shutdown = jest.fn()
+            global.browser = {
+                deleteSession: jest.fn().mockImplementation(() => Promise.reject(new Error('404 session not found'))),
+                sessionId: '123'
+            }
+            expect(runner.endSession()).not.reject
```

Comment:
Why do you expect it not to reject while it does/should? Seems like a false positive.

Reply:
At the time `reloadSession` was "handling" all exceptions raised by `deleteSession`, while `endSession` was not.